pkgs/applications/science/misc/pyfda/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv, python3Packages }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "pyfda";
+  version = "0.2.1";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "11whn2hhr0szbzpmly2p47prvsm2rhizf60iz3pfgkbrnrjs7mhv";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ numpy scipy matplotlib pyqt5 docutils nmigen ];
+
+  patchPhase = ''
+    substituteInPlace setup.py \
+      --replace migen nmigen
+    # The pyfdax_no_term command does not make sense on Linux.
+    substituteInPlace setup.py \
+      --replace "'pyfdax_no_term = pyfda.pyfdax:main'," ""
+
+    substituteInPlace pyfda/tests/test_fixpoint_helpers.py \
+      --replace "from migen import" "from nmigen.compat import"
+    substituteInPlace pyfda/fixpoint_widgets/fir_df.py \
+      --replace "from migen import" "from nmigen.compat import"
+    substituteInPlace pyfda/fixpoint_widgets/fir_df.py \
```

Comment:
```suggestion
```
You can just chain it to the other command, if it's the same file.
However, you make so many changes, it would probably be better to write a patch and add it to `patches = [ ./your-patch.patch ];`.
Because `substituteInPlace` will not fail if the source file has changed to something incompatible; it just prints a warning.